VotingApplication/VotingApplication.Web/Api/Controllers/PollResultsController.cs

```diff
@@ -105,31 +84,46 @@ private static DateTime UnixTimeToDateTime(double unixTimestamp)
             return dateTime;
         }
 
-        private static ResultsRequestResponseModel GenerateResults(IEnumerable<Vote> votes, bool namedVoting)
+        private static ResultsRequestResponseModel GenerateResults(IEnumerable<Vote> votes, IEnumerable<Choice> choices, bool namedVoting)
         {
-            var groupedResults = (from vote in votes
-                                  group vote by vote.Choice.PollChoiceNumber into result
-                                  let optionGroupVotes = result.ToList()
-                                  select new
-                                             {
-                                                 ChoiceName = optionGroupVotes[0].Choice.Name,
-                                                 Sum = optionGroupVotes.Sum(v => v.VoteValue),
-                                                 Voters = optionGroupVotes.Select(v => CreateResultVoteModel(v, namedVoting)).ToList()
-                                             })
-                  .ToList();
-
-            if (!groupedResults.Any())
+            if (choices.Count() == 0)
+            {
+                return new ResultsRequestResponseModel
+                {
+                    Winners = new List<string>(),
+                    Results = new List<ResultModel>()
+                };
+            }
+
+            List<ResultModel> groupedResults = new List<ResultModel>();
+
+            foreach (Choice choice in choices)
             {
-                return new ResultsRequestResponseModel();
+                var choiceVotes = votes.Where(v => v.Choice.Id == choice.Id);
+
+                ResultModel result = new ResultModel();
+                result.ChoiceName = choice.Name;
+                result.Voters = choiceVotes.Select(v => CreateResultVoteModel(v, namedVoting)).ToList();
+                result.Sum = choiceVotes.Sum(v => v.VoteValue);
+
+                groupedResults.Add(result);
             }
 
-            int resultsMax = groupedResults.Max(r => r.Sum);
+            int? resultsMax = groupedResults.Max(r => r.Sum);
 
-            List<string> winners = groupedResults
-                .Where(r => r.Sum == resultsMax)
-                .Select(r => r.ChoiceName)
-                .ToList();
+            List<string> winners;
 
+            if (resultsMax > 0)
```

Comment:
This doesn't feel right. We're comparing `resultsMax`, which is a nullable int to an int, and not checking `hasValue` anywhere. I know that `null` always compares as `false` in this case, but it's _very_ easy to miss the `?` on line 112, and code then implies that we're just working with a regular int, instead of a nullable one.